packages/core/src/test/shared/utilities/pathFind.test.ts

```diff
@@ -42,4 +41,93 @@ describe('pathFind', function () {
         const regex = /bin[\\\/](code|code-insiders)$/
         assert.ok(regex.test(vscPath), `expected regex ${regex} to match: "${vscPath}"`)
     })
+
+    describe('findSshPath', function () {
+        let previousEnv: NodeJS.ProcessEnv
+
+        beforeEach(function () {
+            previousEnv = testutil.readEnv()
+        })
+
+        afterEach(function () {
+            testutil.setEnv(previousEnv)
+        })
+
+        it('first tries ssh in $PATH (Non-Windows)', async function () {
+            // skip on windows because ssh in path will never work with .exe extension.
+            if (isWin()) {
+                return
+            }
+            const workspace = await testutil.createTestWorkspaceFolder()
+            const fakeSshPath = path.join(workspace.uri.fsPath, `ssh`)
+
+            testutil.setEnv(testutil.envWithNewPath(workspace.uri.fsPath))
```

Comment:
if `readEnv` returns a shallow copy, then this becomes much more conventional/familiar:

```suggestion
            process.env.PATH = workspace.uri.fsPath
```